src/Capability/Registry/ElementReference.php

```diff
@@ -12,18 +12,39 @@
 namespace Mcp\Capability\Registry;
 
 /**
- * @phpstan-type Handler \Closure|array{0: object|string, 1: string}|string
+ * Base class for element references with default passthrough argument preparation.
  *
  * @author Kyrian Obikwelu <koshnawaza@gmail.com>
  */
-class ElementReference
+class ElementReference implements ArgumentPreparationInterface
 {
     /**
-     * @param Handler $handler
+     * @param object|array{0: class-string|object, 1: string}|string $handler The handler can be a Closure, array method reference,
+     *                                                                        string function/class name, or a callable object (implementing __invoke)
      */
     public function __construct(
-        public readonly \Closure|array|string $handler,
+        public readonly object|array|string $handler,
```

Comment:
_Note for the reviewer:_ `Closure` `is_a` object, so no BC-break (not that it's a problem at this point...).

I had to do this because now we have the dynamic handlers, that are objects with `__invoke`.